externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/SparkOperation.scala

```diff
@@ -144,7 +151,14 @@ abstract class SparkOperation(session: Session)
           setSessionUserSign()
         }
 
-        f
+        if (isSessionUserAsProxyUser) {
+          UserGroupInformation.createProxyUser(session.user, UserGroupInformation.getCurrentUser)
```

Comment:
Have you tested this solution?

My tests locally are not working as expected.

Create a proxy user. In the Kerberos environment, you should obtain the corresponding proxy user token.

https://github.com/apache/spark/blob/685f536a594700df32c22d62563ba3ec3dd26081/sql/hive/src/main/scala/org/apache/spark/sql/hive/security/HiveDelegationTokenProvider.scala#L104

And regardless of whether there is a Kerberos environment or not, SharedState is only initialized once, which means that the user is determined when the hive meta store client establishes a connection.

https://github.com/apache/spark/blob/685f536a594700df32c22d62563ba3ec3dd26081/sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala#L268-L271

https://github.com/apache/spark/blob/685f536a594700df32c22d62563ba3ec3dd26081/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/IsolatedClientLoader.scala#L324

Reply:
Thanks for your response.

I had tested in a simple auth environment before submitting this PR. It's worked. But you are right; It does not work well above the Kerberos (I confirmed it in my new clusters). 

 I'll try to make sure how it works with  Kerberos in detail. But I'm not very familiar with Kerberos. I'd appreciate it if someone could guide it.